lib/Sanger/CGP/CavemanPostProcessor.pm

```diff
@@ -63,298 +63,278 @@ my $norm_readnames_hash;
 }
 
 sub process_hashed_reads{
-    my ($code, $hashed_reads, $readname_arr) = @_;
-    #Calculate other stuff
-    my $indelRdCount = 0;
-
-    my $loc_counts;
-    $loc_counts->{1}->{A} = 0;
-    $loc_counts->{1}->{C} = 0;
-    $loc_counts->{1}->{G} = 0;
-    $loc_counts->{1}->{T} = 0;
-    $loc_counts->{-1}->{A} = 0;
-    $loc_counts->{-1}->{C} = 0;
-    $loc_counts->{-1}->{G} = 0;
-    $loc_counts->{-1}->{T} = 0;
-
-    foreach my $readnom( @$readname_arr){
-        if(exists $hashed_reads->{$readnom}->{1} && exists $hashed_reads->{$readnom}->{-1}){
-            if($hashed_reads->{$readnom}->{1}->{qbase} ne $hashed_reads->{$readnom}->{-1}->{qbase}){
-                $loc_counts->{1}->{qbase}++;
-                $loc_counts->{1}->{qbase}++;
-            }
-        }elsif(exists $hashed_reads->{$readnom}->{1}){ # + strand populated only
-            $loc_counts->{1}->{qbase}++;
-
-        }else{ # + strand populated only
-            $loc_counts->{-1}->{qbase}++;
-        }
-    } # End of iteration through each readname for an initial count
-
-    foreach my $readnom(@$readname_arr){
-        my $read_to_use;
-        my $read_to_use_2 = undef;
-        if(exists $hashed_reads->{$readnom}->{1} && exists $hashed_reads->{$readnom}->{-1}){
-            if($hashed_reads->{$readnom}->{1}->{qbase} eq $hashed_reads->{$readnom}->{-1}->{qbase}){
-                my $base = $hashed_reads->{$readnom}->{1}->{qbase};
-                if($loc_counts->{-1}->{$base} < $loc_counts->{1}->{$base}){
-                    $read_to_use = $hashed_reads->{$readnom}->{-1};
-                    $loc_counts->{-1}->{$base}++;
-                }else{
-                    $read_to_use = $hashed_reads->{$readnom}->{1};
-                    $loc_counts->{1}->{$base}++;
-                }
-            }else{
-                $read_to_use = $hashed_reads->{$readnom}->{1};
-                $read_to_use_2 = $hashed_reads->{$readnom}->{-1};
-            }
-        }elsif(exists $hashed_reads->{$readnom}->{1}){ # + strand populated only
-            $read_to_use = $hashed_reads->{$readnom}->{1};
-        }else{ # + strand populated only
-            $read_to_use = $hashed_reads->{$readnom}->{-1};
-        }
-
-        &$code($read_to_use);
-        if(defined $read_to_use_2){
-            &$code($read_to_use_2);
-        }
-        delete $hashed_reads->{$readnom};
-    } # End of iteration through each readname for an initial count
-
+  my ($code, $hashed_reads, $readname_arr) = @_;
+  #Calculate other stuff
+  my $indelRdCount = 0;
+
+  my $loc_counts = {
+    1 => {A => 0, C => 0, G => 0, T => 0},
+    -1 => {A => 0, C => 0, G => 0, T => 0},
+  };
+
+  foreach my $readnom(@$readname_arr){
+    my ($read_to_use, $read_to_use_2);
+    if(exists $hashed_reads->{$readnom}->{1} && exists $hashed_reads->{$readnom}->{-1}) {
+    # detected in both take read with the highest qscore
+      if($hashed_reads->{$readnom}->{1}->{qscore} == $hashed_reads->{$readnom}->{-1}->{qscore}) {
```

Comment:
I thought defined behaviour for strand based counting was:
If strands are different, count both. If strands are the same, increment only one strand, 
```suggestion
      if($hashed_reads->{$readnom}->{1}->{qscore} == $hashed_reads->{$readnom}->{-1}->{qscore}) {
```
```suggestion
      if($hashed_reads->{$readnom}->{1}->{qscore} == $hashed_reads->{$readnom}->{-1}->{qscore}) {
```

Reply:
That was the original intent but before the extra block where you attempted to balance  +/- contribution.

My original attempt only fixed the issue with both being assigned to the +ve strand and that it only ever was set for a disagreement if reads overlapped (the reason we end up with it failing).  It had no logic at all that handled when the reads agree.

https://github.com/cancerit/cgpCaVEManPostProcessing/pull/36/commits/d73ed23e3728d2d555bc1c644c5fbe71de2c1b8a

I subsequently switched to the base qual as the second block (deleted) attempts to balance the +/-ve contribution, thus no longer double counting anything.

https://github.com/cancerit/cgpCaVEManPostProcessing/pull/36/commits/536d59cb10467a06aea19b124afd01288e4ccf21#diff-8614509bb785fd3a892d6da53278dc2bL311-L321

Reply:
Still takes both reads if the qscore is the same.